### PR TITLE
Document how to default struct fields in CRD

### PR DESCRIPTION
**Pull request by gibizer**
Thanks to @stuggi we have a solution for properly defaulting struct fields.

### CODE REVIEW

File: developer.md [hunk 1]
```diff
@@ -131,3 +131,38 @@ There are two reasons to add a finalizer to a CR instance from its reconciler:
...
+special care. Both the field with the struct type need a full default value
+defined and each individual subfields in the struct needs default value
+defined.
+
+Our common example is the `PasswordSelectors` struct.
+
+```golang
+type PlacementAPISpec struct {
+    //...
+    // +kubebuilder:validation:Optional
```
Comment (stuggi) [inline developer.md:146]: maybe fix this to align with the next line?

**Review by stuggi — COMMENTED** [review]
thanks for documenting this Gibi, I think just the indentation of the Optional validation should align with the default?

File: developer.md (hunk 1 above)
Comment (gibizer) [reply developer.md]: I blame vscode not showing the tab and the spaces differently. Now I replaced all with tabs.

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**